src/config/ConfigWatcher.cpp

```diff
@@ -43,9 +44,16 @@ void CConfigWatcher::setWatchList(const std::vector<std::string>& paths) {
     // add new paths
     for (const auto& path : paths) {
         m_watches.emplace_back(SInotifyWatch{
-            .wd   = inotify_add_watch(m_inotifyFd.get(), path.c_str(), IN_MODIFY),
+            .wd   = inotify_add_watch(m_inotifyFd.get(), path.c_str(), IN_MODIFY | IN_DONT_FOLLOW),
             .file = path,
         });
+        std::error_code ec;
+        if (const auto CANONICAL = std::filesystem::canonical(path, ec); !ec) {
```

Comment:
you should also check that it is a symlink at all, otherwise you get 2 watches for one file for no reason